examples/black_scholes_greeks.py

```diff
@@ -0,0 +1,217 @@
+            n_t_steps,
+            n_vol_steps,
+            n_money_steps,
+        ),
+        dtype=D,
+    )
+    PUT = np.zeros(
+        (
+            N_GREEKS,
+            n_t_steps,
+            n_vol_steps,
+            n_money_steps,
+        ),
+        dtype=D,
+    )
+    S = np.full(
+        (
+            n_t_steps,
+            n_vol_steps,
+            n_money_steps,
+        ),
+        S0,
+        dtype=D,
+    )
+    temp_arr = np.arange((n_vol_steps * n_t_steps * n_money_steps), dtype=int)
+    k_temp = (temp_arr % n_money_steps) * money_step
+    k_temp = k_temp.reshape(
+        (
+            n_t_steps,
+            n_vol_steps,
+            n_money_steps,
+        )
+    )
+    K = (k_temp + (1 + money_start)) * S0
+
+    t_temp = (temp_arr % (n_vol_steps * n_money_steps)) * vol_step
+    t_temp = t_temp.reshape(
+        (
+            n_t_steps,
+            n_vol_steps,
+            n_money_steps,
+        )
+    )
+    T = t_temp + t_start
+    R = 0.02
+    V = np.full((n_t_steps, n_vol_steps, n_money_steps), vol_start, dtype=D)
+    for i in range(n_vol_steps):
+        V[:, i, :] += i * vol_step
+
+    return CALL, PUT, S, K, T, R, V
+
+
+# Cumulative distribution function
+# https://en.wikipedia.org/wiki/Cumulative_distribution_function
+def normCDF(d):
+    A1 = 0.31938153
+    A2 = -0.356563782
+    A3 = 1.781477937
+    A4 = -1.821255978
+    A5 = 1.330274429
+    RSQRT2PI = 0.39894228040143267793994605993438
+
+    K = 1.0 / (1.0 + 0.2316419 * np.absolute(d))
+
+    cnd = (
+        RSQRT2PI
+        * np.exp(-0.5 * d * d)
+        * (K * (A1 + K * (A2 + K * (A3 + K * (A4 + K * A5)))))
+    )
+
+    return np.where(d > 0, 1.0 - cnd, cnd)
+
+
+# Probability density function.
+# https://en.wikipedia.org/wiki/Probability_density_function
+def normPDF(d):
+    RSQRT2PI = 0.39894228040143267793994605993438
+    return RSQRT2PI * np.exp(-0.5 * d * d)
+
+
+# https://en.wikipedia.org/wiki/Black–Scholes_model
+def black_scholes(out, S, K, R, T, V, CP, greek):
+    stdev = V * np.sqrt(T)
+    df = np.exp(-R * T)
+    d1 = (np.log(S / K) + (R + 0.5 * V * V) * T) / stdev
+    d2 = d1 - stdev
+    nd1 = normCDF(CP * d1)
+    nd2 = normCDF(CP * d2)
+
+    if greek == Greeks.PREM:
+        out[...] = CP * (S * nd1 - K * df * nd2)
+    elif greek == Greeks.DELTA:
+        out[...] = CP * nd1
+    elif greek == Greeks.VEGA:
+        out[...] = S * np.sqrt(T) * normPDF(d1)
+    elif greek == Greeks.GAMMA:
+        out[...] = normPDF(d1) / (S * V * np.sqrt(T))
+    elif greek == Greeks.VANNA:
+        out[...] = -d2 * normPDF(d1) / V
+    elif greek == Greeks.VOLGA:
+        out[...] = S * np.sqrt(T) * d1 * d2 * normPDF(d1) / V
+    elif greek == Greeks.THETA:
+        out[...] = -(
+            0.5 * S * V / np.sqrt(T) * normPDF(d1) + CP * R * df * K * nd2
+        )
+    else:
+        raise RuntimeError("Wrong greek name is passed")
+
+
+def run_black_scholes(n_vol_steps, n_t_steps, n_money_steps):
```

Comment:
Maybe `run_black_scholes_benchmark` (or similar)? This function runs B-S across all combinations of put/call and greeks and computes the average time, including handling warmup iterations.